idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -536,127 +536,125 @@ def initialize(self, state_args={}, outlvl=idaeslog.NOTSET, solver='ipopt',
             solver (str): Solver to use for initialization
             optarg (dict): Solver arguments dictionary
         """
-        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="unit")
-        solve_log = idaeslog.getSolveLogger(self.name, outlvl, tag="unit")
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
 
-        opt = SolverFactory(solver)
-        opt.options = optarg
-
-        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
-            res = opt.solve(self, tee=slc.tee)
-        init_log.info_high(
-                "Initialization Step 1 {}.".format(idaeslog.condition(res))
-            )
+        flags = blk.control_volume.initialize(state_args=state_args,
+                                              hold_state=True)
 
-        for t in self.flowsheet().config.time:
-            for k, v in self.inlet.vars.items():
+        for t in blk.flowsheet().config.time:
+            for k, v in blk.inlet.vars.items():
                 v[t].fix()
-            for k, v in self.outlet.vars.items():
+            for k, v in blk.outlet.vars.items():
                 v[t].unfix()
 
             # if there is not a good guess for efficiency or outlet pressure
             # provide something reasonable.
-            eff = self.efficiency_isentropic[t]
+            eff = blk.efficiency_isentropic[t]
             eff.fix(eff.value if value(eff) > 0.3
                     and value(eff) < 1.0 else 0.85)
             # for outlet pressure try outlet pressure, pressure ratio
             # then if none of those look reasonable use a pressure ratio of 2.0
             # to calculate outlet pressure
-            Pout = self.outlet.pressure[t]
-            Pin = self.inlet.pressure[t]
+            Pout = blk.outlet.pressure[t]
+            Pin = blk.inlet.pressure[t]
             prratio = value(Pout/Pin)
             if value(Pout / Pin) < 4.0 or value(Pout / Pin) > 1.0:
-                if value(self.ratioP[t]) < 4.0 and value(self.ratioP[t]) > 1.0:
-                    Pout.fix(value(Pin * self.ratioP[t]))
+                if value(blk.ratioP[t]) < 4.0 and value(blk.ratioP[t]) > 1.0:
+                    Pout.fix(value(Pin * blk.ratioP[t]))
                 elif prratio < 4.0 and prratio > 1.0:
                     Pout.fix(value(prratio*Pin))
                 else:
                     Pout.fix(value(Pin * 2.0))
             else:
                 Pout.fix()
 
-        self.deltaP[:] = value(Pout - Pin)
-        self.ratioP[:] = value(Pout / Pin)
+        blk.deltaP[:] = value(Pout - Pin)
```

Comment:
This is not good, as it will overwrite any user provided value for deltaP and ratioP - i.e. what if the user fixed one of htese values instead of Pout?